lib/storage/filesystem.js

```diff
@@ -49,22 +49,43 @@ function save(file, db, cb) {
 	if (writeQueue[file]) {
 		writeQueue[file].db = db;
 		writeQueue[file].callbacks.push(cb);
+        writeQueue[file].version += 1;
 		return;
 	} else {
-		writeQueue[file] = { callbacks: [cb], db };
+		writeQueue[file] = { callbacks: [cb], db,  version: 1};
 	}
 
-	// If not already saving then save, but using the object we make sure to take only
-	// the last changes.
-	// Truncate - https://stackoverflow.com/questions/35178903/overwrite-a-file-in-node-js
-	fs.truncate(file, err => {
-		fse.outputJson(file, writeQueue[file].db, err => {
-			resolveAllWriteQueueCallbacks(file, err);
-			delete writeQueue[file];
-		});
-	});
+    writeToFile(file);
 }
 
+/**
+ * Save db data to a file from the write queue.
+ * 
+ * @param {string} file File path.
+ * @returns {void} Nothing
+ */
+function writeToFile(file) {
+	if (writeQueue[file] === undefined) {
```

Comment:
Can we use `!writeQueue[file]`?

Reply:
I always though this way is faster because it has to test just undefined and it shouldn't have any other falsy value. Am I wrong?